### PR TITLE
Feat/5/3 Transaction Waiter

### PR DESCRIPTION
**Pull request by piniom**
In this PR I've rewritten the TransactionWaiter to use async/await. It should now also compile to wasm.

### CODE REVIEW

File: crates/account_sdk/Cargo.toml [hunk 1]
```diff
@@ -30,5 +30,10 @@ toml.workspace = true
 u256-literal.workspace = true
 url.workspace = true
 wasm-bindgen.workspace = true
-webauthn-authenticator-rs.workspace = true
 webauthn-rs-proto.workspace = true
+
```
Comment (JunichiSugiura) [inline crates/account_sdk/Cargo.toml:34]: i think tokio dep has to be excluded from wasm32 target

```toml
[target.'cfg(not(target_arch = "wasm32"))'.dependencies]
tokio.workspace = true
```

Comment (piniom) [reply crates/account_sdk/Cargo.toml:33]: I've fixed this, thank you.

File: README.md [hunk 2]
```diff
@@ -16,14 +16,30 @@ The project has a global rust workspace.
...
 
+in the root directory.
+
+### Building for web assembly
+
+After you've compiled the cairo code you can compile rust to wasm using
+
+```bash
+cd crates/account_sdk
+cargo build --target wasm32-unknown-unknown  --release
```
Comment (JunichiSugiura) [inline README.md:36]: I prefer `-p account-sdk` flag for one-liner execution.

Comment (piniom) [reply README.md]: Yup, fixed that

**Comment by tarrencev** [comment]
Merging but we should move to the Dojo waiter impl in the future

**Pull request merged**